SafeGraph_py_functions/SafeGraph_py_functions.py

```diff
@@ -113,65 +113,99 @@ def unpack_json_separate(df_, json_column='visitor_home_cbgs', key_col_name='vis
+        df['index_original'] = df.index
+    df.reset_index(drop=True, inplace=True) # THIS IS IMPORTANT; explode will not work correctly if index is not unique
+    df[array_column+'_json'] = [json.loads(myjson) for myjson in df[array_column]]
+    day_visits_exp = df[[place_key, file_key, array_column+'_json']].explode(array_column+'_json')
+    day_visits_exp['dummy_key'] = day_visits_exp.index
+    day_visits_exp[array_sequence] = day_visits_exp.groupby([place_key, file_key])['dummy_key'].rank(method='first', ascending=True).astype('int64')
+    if(zero_index):
+      day_visits_exp[array_sequence] = day_visits_exp[array_sequence] -1
+    day_visits_exp.drop(['dummy_key'], axis=1, inplace=True)
+    day_visits_exp.rename(columns={array_column+'_json': new_col}, inplace=True)
+    day_visits_exp[new_col] = day_visits_exp[new_col].astype('int64')
+    df.drop([array_column+'_json'], axis=1, inplace=True)
+    df = pd.merge(df, day_visits_exp, on=[place_key,file_key])
+    return df
 
 ### ------------------------------------------ END JSON SECTION--------------------------------------------------------
 
-### ---------------------------------------CORE AND PATTERNS SECTION -----------------------------------------------
+### ---------------------------------------CORE, GEO, AND PATTERNS SECTION -----------------------------------------------
 
 
-def read_core_folder(path_to_core, use_cols=None):
+def read_core_folder(path_to_core, *args, **kwargs):
     core_files = glob.glob(os.path.join(path_to_core, "*.csv.gz"))
     print(f"You are about to load in {len(core_files)} core files")
 
     li = []
     for core in core_files:
         print(core)
-        df = pd.read_csv(core, usecols=use_cols, compression='gzip',
+        df = pd.read_csv(core, *args, **kwargs, compression='gzip',
                          dtype={'postal_code': str, 'phone_number': str, 'naics_code': str})
         li.append(df)
 
     SG_core = pd.concat(li, axis=0)
     return SG_core
 
+### added a new core read that takes the information straight from the zipped file (like you get it from the catelog)
 
-def read_pattern_demo(f_path, use_cols=None, compression='gzip', nrows=100):
-    df = pd.read_csv(f_path, dtype={'postal_code': str, 'phone_number': str, 'naics_code': str}, nrows=nrows,
-                     compression=compression,
-                     usecols=use_cols)
-    return df
+def read_core_folder_zip(path_to_core, *args, **kwargs):
+    zip_file = ZipFile(path_to_core)
+
+    li = []
+
+    dfs = {text_file.filename: pd.read_csv(zip_file.open(text_file.filename), *args, **kwargs, compression='gzip',
+                         dtype={'postal_code': str, 'phone_number': str, 'naics_code': str})
+           for text_file in zip_file.infolist()
+           if text_file.filename.endswith('.csv.gz')}
+
+    SG_core = pd.concat(dfs, axis=0, ignore_index=True)
+
+    return SG_core
+
+def read_geo_zip(path_to_zip, *args, **kwargs):
+  zf = zp.ZipFile(path_to_zip)
```

Comment:
@Trippl7777  on line 168 you use just `ZipFile` and at the top you imported both ZipFile as well as zp. I think we only need one or the other, right? no need for zp?

Reply:
Yeah, sorry about that, I was using it for a second ZipFile function. I am not actually loading in all of zipfile for the first one, but I will try to get it down to one line of import